tests/run/typelevel.scala

```diff
@@ -0,0 +1,31 @@
+object Test extends App {
```

Comment:
Maybe give this file and its '.check' companion a more descriptive name like `transparent.scala` or `TransparentToNat.scala`?

Reply:
I planned to extend this test with more typelevel operations as we go along.